.github/workflows/pull-request.yml

```diff
@@ -31,6 +31,16 @@ jobs:
         with:
           version: ${{ env.BAL_VERSION }}
 
+      - name: Set up Maven
+        uses: stCarolas/setup-maven@v4
+        with:
+          maven-version: '3.9.6'
```

Comment:
We don't need to set up Maven separately. We can try passing maven as an option when setting up java like

```
      - name: Set up JDK 21
        uses: actions/setup-java@v3
        with:
          distribution: 'temurin'
          java-version: 21.0.3
          cache: maven
```